ssl/ssl_lib.c

```diff
@@ -2262,6 +2262,11 @@ int SSL_key_update(SSL *s, int updatetype)
         return 0;
     }
 
+    if (RECORD_LAYER_write_pending(&s->rlayer)) {
```

Comment:
Hmm, should the check in `SSL_new_session_ticket()` use `RECORD_LAYER_write_pending()` instead of peeking into `s->rlayer.wbuf[0].left`?

Reply:
```Also, the comment at the top of the body of ssl3_read_n() refers to s->packet and s->packet_length, which seems out of date with respect to the RECORD_LAYER abstraction.``` The following comments have the same problem.
https://github.com/openssl/openssl/blob/11f18ef1f5ea92ba32a3efeb0eaf1af6a1f35e13/ssl/record/ssl3_record.c#L430-L446

Reply:
Yes. It probably should (different PR though)